LanguageFeatures/Augmentation-libraries/augmenting_functions_A06_t01.dart

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It is a compile-time error if:
+/// ...
+/// - An augmenting declaration uses augmented when the original declaration has
+///   no concrete implementation. Note that all external declarations are
+///   assumed to have an implementation provided by another external source, and
+///   they will throw a runtime exception when called if not.
+///
+/// @description Checks that it is a compile-time error if an augmenting
+/// declaration uses `augmented` when the original declaration has no concrete
+/// implementation.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+import augment 'augmenting_functions_A06_t01_lib.dart';
+
+class C1 {
+  void foo();
```

Comment:
This is interesting!

We could say that nothing except syntax errors counts as a compile-time error in the main library or in any library augmentation, we're _always_ waiting to see whether there is an error in the final, merged library.

That would be sufficient to accept this declaration, assuming that it would receive an implementation from an augmentation. We could claim that this doesn't happen after all because the augmentation has an error; however, I think it makes sense to say that we do receive an implementation of `foo` from an augmentation, and it's a separate issue that it has an error because it uses `augmented`.

OK, that's good!